doc/website/blog/2021-11-10-announcing-entoas.md

```diff
@@ -0,0 +1,270 @@
+over to the [Setup Tutorial](https://entgo.io/docs/tutorial-setup/).
+
+### Generate an OAS document
+
+The first step on our way to the OAS document is to create an Ent schema graph. For the sake of brevity here is an
+example schema to use:
+
+Let's create our schema:
+
+```go title="ent/schema/schema.go"
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+)
+
+// Fridge holds the schema definition for the Fridge entity.
+type Fridge struct {
+	ent.Schema
+}
+
+// Fields of the Fridge.
+func (Fridge) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("title"),
+	}
+}
+
+// Edges of the Fridge.
+func (Fridge) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.To("compartments", Compartment.Type),
+	}
+}
+
+// Compartment holds the schema definition for the Compartment entity.
+type Compartment struct {
+	ent.Schema
+}
+
+// Fields of the Compartment.
+func (Compartment) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name"),
+	}
+}
+
+// Edges of the Compartment.
+func (Compartment) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("fridge", Fridge.Type).
+			Ref("compartments").
+			Unique(),
+		edge.To("contents", Item.Type),
+	}
+}
+
+// Item holds the schema definition for the Item entity.
+type Item struct {
+	ent.Schema
+}
+
+// Fields of the Item.
+func (Item) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("name"),
+	}
+}
+
+// Edges of the Item.
+func (Item) Edges() []ent.Edge {
+	return []ent.Edge{
+		edge.From("compartment", Compartment.Type).
+			Ref("contents").
+			Unique(),
+	}
+}
+```
+
+Now run the code generator:
+
+```shell
+go generate ./...
+```
+
+In addition to the files Ent normally generates, another file named `ent/openapi.json` has been created. Copy its contents
+and paste them into the [Swagger Editor](https://editor.swagger.io/). You should see three groups: **Compartment**, **
+Item** and **Fridge**.
+
+### Basic Configuration
+
+The description of our API does not yet reflect what it does, but `entoas` lets you change that! Open up `ent/entc.go`
+and pass in the updated title and description of our Fridge API:
+
+```go title="ent/entc.go"
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"log"
+
+	"entgo.io/contrib/entoas"
+	"entgo.io/ent/entc"
+	"entgo.io/ent/entc/gen"
+)
+
+func main() {
+	ex, err := entoas.NewExtension(
+		entoas.SpecTitle("Fridge CMS"),
+		entoas.SpecDescription("API to manage fridges and their cooled contents. **ICY!**"), 
+		entoas.SpecVersion("0.0.1"),
+	)
+	if err != nil {
+		log.Fatalf("creating entoas extension: %v", err)
+	}
+	err = entc.Generate("./schema", &gen.Config{}, entc.Extensions(ex))
+	if err != nil {
+		log.Fatalf("running ent codegen: %v", err)
+	}
+}
+```
+
+Rerunning the code generator will create an updated OAS document you can copy-paste into the Swagger Editor.
```

Comment:
again, show don't tell. show a snippet of the JSON file that shows this change. 
always show cause and effect so people can follow along by example instead